File: frontend/src/services/openapi/models/CurrentCompetitorEnvelopeModel.ts
/* istanbul ignore file */
/* tslint:disable */
/* eslint-disable */

import type { CurrentCompetitorContentModel } from './CurrentCompetitorContentModel';

/**
 * Envelope for current competitor. Current competitor is the competitor that
 * is performing currently or is performing next, if no one is performing.
 */
export type CurrentCompetitorEnvelopeModel = {
    /**
     * Envelope version number. Version can be discarded if no
     * functionality is specified for given version
     */
    readonly version?: string | null;
    /**
     * Type of the message. This and version can be used to identify
     * correct parser for this message.
     */
    readonly type?: string | null;
    content?: CurrentCompetitorContentModel;
}

File: src/Api.Models/CurrentCompetitorEnvelopeModel.cs
namespace Api.Models;

/// <summary>
/// Envelope for current competitor. Current competitor is the competitor that
/// is performing currently or is performing next, if no one is performing.
/// </summary>
public class CurrentCompetitorEnvelopeModel : BaseEnvelopeModel<CurrentCompetitorContentModel>
{
    public CurrentCompetitorEnvelopeModel()
    {
        Version = "1";
        Type = "current-competitor";
    }
}
